github3/repos/repo.py

```diff
@@ -620,6 +620,30 @@ def create_blob(self, content, encoding):
         return sha
 
     @requires_auth
+    def create_branch(self, name, sha=None):
+        """Create a branch. Returns False if branch doesn't exist
+        or not authorized.
+
+        :param str branch: (required), the branch to create.
+        :param str|commit sha: (optional), the commit to base
```

Comment:
I *really* don't want to make an extra API call for any method we don't need to. Users can certainly do `repo.commit('HEAD')` themselves. I'm perfectly okay with accepting a `github3.git.Commit` object though if you add the `__str__` method to it so it returns `self.sha`. This means `create_ref` only needs to do:

```py
sha = str(sha)
```

Finally, you need to update the doc-string here. `create_ref` does not return a boolean, and we won't be making the extra API call to get the value from `HEAD`.